### PR TITLE
[MDS-6170] Update major project table to reflect project statuses

### PR DESCRIPTION
**Pull request by asinn134**
## Objective 

[MDS-6170](https://bcmines.atlassian.net/browse/MDS-6170)

- Updated the all major projects table to give a more accurate view on the status of the project stages. Added ability to expand project row to view a sub table with project stage status info.

- Removed MRC, Stage, Status columns from the table. Also removed the same from the search filters.

- Made default order of table to be descending and sorted by updated date.

- Converted some older JS files to Typescript.

Screenshot (510)

### CODE REVIEW

File: services/common/src/redux/actionCreators/projectActionCreator.ts [hunk 1]
```diff
@@ -234,7 +234,7 @@ export const fetchProjectById = (projectGuid: string): AppThunk<Promise<IProject
 };
 
 export const fetchProjects = (params: IProjectPageData): AppThunk => (dispatch) => {
-  const defaultParams = params || Strings.DEFAULT_DASHBOARD_PARAMS;
+  const defaultParams = params || `${Strings.DEFAULT_DASHBOARD_PARAMS}${Strings.DEFAULT_MAJOR_PROJECTS_DASHBOARD_PARAMS}`;
```
Comment (asinn134) [inline services/common/src/redux/actionCreators/projectActionCreator.ts:237]: This is the only change in this file

File: services/core-web/src/components/Forms/MajorProject/MajorProjectSearchForm.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import React, { FC } from "react";
+import { UpOutlined, DownOutlined } from "@ant-design/icons";
+import { Field, reduxForm } from "redux-form";
+import { Form } from "@ant-design/compatible";
+import "@ant-design/compatible/assets/index.css";
+import { Button, Col, Row } from "antd";
+
+import * as FORM from "@/constants/forms";
+import { renderConfig } from "@/components/common/config";
```
Comment (taraepp) [inline services/core-web/src/components/Forms/MajorProject/MajorProjectSearchForm.tsx]: how much work would it be to take out the old inputs and the old antd compatible Form/css file?

File: services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  const toggleIsAdvancedSearch = () => setExpandAdvancedSearch(!expandAdvancedSearch);
+
+  return (
+    <div>
+      <Row>
+        <Col md={{ span: 12, offset: 6 }} xs={{ span: 20, offset: 2 }}>
+          <span className="advanced-search__container">
+            <MajorProjectSearchForm
+              // @ts-ignore
```
Comment (taraepp) [inline services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectSearch.tsx]: I'm curious what this is ts-ignoring

File: services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+import { Link } from "react-router-dom";
+import { Button, Col, Row } from "antd";
+import { uniqBy, flattenDeep } from "lodash";
+import * as Strings from "@mds/common/constants/strings";
+import {
+  PROJECT_SUMMARY_STATUS_CODES,
+  MAJOR_MINE_APPLICATION_AND_IRT_STATUS_CODES,
+  PROJECT_DECISION_PACKAGE_STATUS_CODES
+} from "@mds/common/constants/enums";
+import { formatDate, dateSorter } from "@common/utils/helpers";
```
Comment (taraepp) [inline services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx]: this one I'd recommend updating the import to be from @mds/common

File: services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+      sorter: dateSorter("update_timestamp"),
+      defaultSortOrder: "descend",
+    },
+    {
+      title: "",
+      key: "project",
+      dataIndex: "project",
+      render: (text, record) => (
+        // @ts-ignore
+        <div title="" align="right" data-cy="major-projects-table-open-button">
```
Comment (taraepp) [inline services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx]: does the `align="right"` even work on div? Or does it just want to be corrected to `style={{align:"right"}}`?

File: services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+  const childColumns = [
+    { ...renderTextColumn("key", "Project Stage") },
+    { ...renderTextColumn("status", "EMLI Review Status") },
+    { ...renderTextColumn("updateTimestamp", "Last Updated") },
+    { ...renderTextColumn("updateUser", "Updated By") },
+    {
+      title: "",
+      key: "project_section",
+      dataIndex: "project_section",
+      render: (text, record) => {
```
Comment (taraepp) [inline services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx]: Looks like the render method doesn't use the `text` parameter (which will be `record.project_section`). If that's the case- you can take out the dataIndex and the text parameter completely.

File: services/core-web/src/tests/components/dashboard/majorProjectHomePage/MajorProjectSearch.spec.js [hunk 7]
```diff
@@ -1,7 +1,8 @@
 import React from "react";
-import { shallow } from "enzyme";
 import MajorProjectSearch from "@/components/dashboard/majorProjectHomePage/MajorProjectSearch";
 import * as MOCK from "@/tests/mocks/dataMocks";
```
Comment (taraepp) [inline services/core-web/src/tests/components/dashboard/majorProjectHomePage/MajorProjectSearch.spec.js]: may as well update this import to be from @mds/common. It has more complete data and will give more useful results. Same for the other test below.

File: services/core-web/src/tests/mocks/dataMocks.tsx [hunk 8]
```diff
@@ -6725,6 +6720,26 @@ export const MAJOR_PROJECTS_DASHBOARD = {
...
+      project_decision_package: {
+        status_code: "SUB",
+        update_timestamp: "2022-08-01T00:00:00+00:00",
+        update_user: "Joe Average",
+      },
+      project_summary: {
+        status_code: "SUB",
+        update_timestamp: "2022-08-01T00:00:00+00:00",
+        update_user: "Joe Average",
+      },
```
Comment (taraepp) [inline services/core-web/src/tests/mocks/dataMocks.tsx:6742]: yeah definitely just copy these changes over to the common file.

File: services/core-web/src/components/Forms/MajorProject/MajorProjectSearchForm.tsx (hunk 2 above)
Comment (asinn134) [reply services/core-web/src/components/Forms/MajorProject/MajorProjectSearchForm.tsx]: The old antd form has been taken out from here now.

File: services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectSearch.tsx (hunk 3 above)
Comment (asinn134) [reply services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectSearch.tsx]: This has been removed now.

File: services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx (hunk 4 above)
Comment (asinn134) [reply services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx]: The import has been changed to be from @mds/common now

File: services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx (hunk 5 above)
Comment (asinn134) [reply services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx]: It looks like it doesn't work, I just removed this and also the ts-ignore.

File: services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx (hunk 6 above)
Comment (asinn134) [reply services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx]: Thanks this has been removed now.

File: services/core-web/src/tests/components/dashboard/majorProjectHomePage/MajorProjectSearch.spec.js (hunk 7 above)
Comment (asinn134) [reply services/core-web/src/tests/components/dashboard/majorProjectHomePage/MajorProjectSearch.spec.js]: The dataMocks has been changed to be from @mds/common here too.

File: services/core-web/src/components/Forms/MajorProject/MajorProjectSearchForm.tsx (hunk 2 above)
Comment (taraepp) [reply services/core-web/src/components/Forms/MajorProject/MajorProjectSearchForm.tsx]: Can the `renderConfig` go too? Those are old antd inputs. renderConfig.Field for example can just be replaced with RenderField (imported from @mds/common).

File: services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+    { ...renderTextColumn("updateTimestamp", "Last Updated") },
+    { ...renderTextColumn("updateUser", "Updated By") },
+    {
+      title: "",
+      key: "project_section",
+      render: (record) => {
+        return (
+          // @ts-ignore
+          <div title="" align="right">
+            <Row gutter={1}>
```
Comment (taraepp) [inline services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx:174]: there's another one

Comment (asinn134) [reply services/core-web/src/components/dashboard/majorProjectHomePage/MajorProjectTable.tsx:174]: Thanks! this has been removed now.

File: services/core-web/src/components/Forms/MajorProject/MajorProjectSearchForm.tsx (hunk 2 above)
Comment (asinn134) [reply services/core-web/src/components/Forms/MajorProject/MajorProjectSearchForm.tsx]: `renderConfig` has been replaced with the Render options from @mds/common now.

**Review by taraepp — APPROVED** [review]

**Review by matbusby-fw — APPROVED** [review]
Nice!  This looks great!

**Pull request merged**